repose-aggregator/src/docs/asciidoc/filters/api-validator.adoc

```diff
@@ -11,45 +26,283 @@ include::../_includes/in-progress.adoc[]
+|===
 
 == Examples
-=== Shows how to ???
-This configuration will ???.
+=== Basic Example
+This configuration is the most basic example of this filter's configuration.
+
+[source,xml]
+.validator.cfg.xml
+----
+<validators xmlns="http://docs.openrepose.org/repose/validator/v1.0">
+    <validator role="default" <!--1-->
+               default="true" <!--2-->
+               wadl="file:///path/to/wadl/file.wadl" <!--3-->
+    />
+</validators>
+----
+<1> The `role` attribute is deprecated, but is currently required and doesn't have a default value.
+<2> The `default` attribute is deprecated, but currently has a default value of `false`.
+<3> This is the fully qualified URI for where the WADL file can be located.
+
+[NOTE]
+====
+Even though multiple `validator` element definitions and the `role` and `default` attributes on each are deprecated, the easiest way to future proof new configurations is to:
+
+* define a single `validator` as this will eventually become the root element
+* define that `validator` element with:
+** a `role` attribute with the value of `default`
+** a `default` attribute with the value of `true`
+====
+
+=== All the features of a `validator` element
+This configuration expands the basic example in order to show off all of the features of this element.
+
+[source,xml]
+.validator.cfg.xml
+----
+<validators xmlns="http://docs.openrepose.org/repose/validator/v1.0">
+    <validator role="default" <!--1-->
+               default="true" <!--2-->
+               wadl="file:///path/to/wadl/file.wadl" <!--3-->
+               enable-api-coverage="false" <!--4-->
+               dot-output="/tmp/default.dot" <!--5-->
+               check-well-formed="false" <!--6-->
+               check-grammars="false" <!--7-->
+               check-elements="true" <!--8-->
+               check-plain-params="true" <!--9-->
+               do-xsd-grammar-transform="true" <!--10-->
+               enable-pre-process-extension="true" <!--11-->
+               remove-dups="true" <!--12-->
+               xpath-version="2" <!--13-->
+               xsl-engine="XalanC" <!--14-->
+               xsd-engine="Xerces" <!--15-->
+               enable-ignore-xsd-extension="false" <!--16-->
+               join-xpath-checks="false" <!--17-->
+               validator-name="testName" <!--18-->
+               check-headers="true" <!--19-->
+               enable-rax-roles="false" <!--20-->
+               mask-rax-roles-403="false" <!--21-->
+               validate-checker="true" <!--22-->
+    />
+</validators>
+----
+<1> List of roles that are applied on single validator.
+    Triggers off of `X-Roles` header. +
+    *DEPRECATED:* Roles defined outside of the WADL will not be supported in Repose 9 and this attribute will not be available.
+<2> Set to `true` to use this validator if no `X-Roles` header is passed.
```

Comment:
Suggestion:


> Set to `true` to use this validator if no other validator applies, or if `multi-match` is set to `true`.

As written, it is not clear that this validator will be applied even if the `X-Roles` header is present, but does not contain any matching roles. That is clarified a couple of lines down, but it would be better if we could avoid the ambiguity in the first place.

Alternatively, you could just drop this line and rely on the two following lines instead.